mkosi/__init__.py

```diff
@@ -608,22 +562,8 @@ def configure_serial_terminal(state: MkosiState, cached: bool) -> None:
                           """)
 
 
-def nspawn_id_map_supported() -> bool:
-    if nspawn_version() < 252:
-        return False
-
-    ret = run(["systemd-analyze", "compare-versions", platform.release(), ">=", "5.12"], check=False)
-    return ret.returncode == 0
-
-
-def nspawn_params_for_build_sources(config: MkosiConfig, sft: SourceFileTransfer) -> list[str]:
-    params = ["--setenv=SRCDIR=/root/src",
-              "--chdir=/root/src"]
-    if sft == SourceFileTransfer.mount:
-        idmap_opt = ":rootidmap" if nspawn_id_map_supported() and config.idmap else ""
-        params += [f"--bind={config.build_sources}:/root/src{idmap_opt}"]
-
-    return params
+def script_env_params(state: MkosiState) -> list[str]:
+    return flatten(("--setenv", k, v) for k, v in state.environment.items())
```

Comment:
This should be moved to `run.py`, so `__init__.py` gets even emptier.

Reply:
Moved to backend, I think it fits better there